src/dolphin/atmosphere/troposphere.py

```diff
@@ -0,0 +1,557 @@
+from __future__ import annotations
+
+import datetime
+import os
+from dataclasses import dataclass
+from os import fspath
+from pathlib import Path
+from typing import Literal, Optional, Union
+
+import numpy as np
+import opera_utils as oput
+import pyaps3 as pa
+from osgeo import gdal
+from pyproj import CRS
+from scipy.interpolate import RegularGridInterpolator
+
+from dolphin import io, stitching
+from dolphin._dates import _format_date_pair, get_dates
+from dolphin._log import get_log
+from dolphin._types import Bbox, Filename
+
+logger = get_log(__name__)
+
+
+###########
+
+
+@dataclass
+class DelayParams:
+    """Parameters for estimating tropospheric delay corrections.
+
+    Attributes
+    ----------
+        x_coordinates (np.ndarray): X coordinates.
+        y_coordinates (np.ndarray): Y coordinates.
+        z_coordinates (np.ndarray): Z coordinates.
+        bounds (Bbox): Bounding box.
+        shape (tuple[int, int]): Shape of the data.
+        epsg (int): EPSG code for the coordinate reference system.
+        geotransform (list[float]): Geotransformation parameters.
+        wavelength (float): Radar wavelength.
+        tropo_model (Union[str, Literal["ERA5", "HRES", "ERAINT", "ERAI", "MERRA", "NARR", "HRRR", "GMAO"]]):
+            Tropospheric model.
+        delay_type (Union[str, Literal["wet", "dry", "hydrostatic", "comb"]]):
+            Type of tropospheric delay.
+        reference_file (List[Filename]): List of reference files.
+        secondary_file (List[Filename]): List of secondary files.
+        interferogram (str): Interferogram identifier.
+        reference_time (datetime.datetime): Reference time.
+        secondary_time (datetime.datetime): Secondary time.
+        delay_datacube (Optional[np.ndarray]): Optional delay data cube.
```

Comment:
making it compliant with https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring


```suggestion
    x_coordinates : np.ndarray
        Array of X coordinates.
    y_coordinates : np.ndarray
        Array of Y coordinates.
    z_coordinates : np.ndarray
        Array of Z coordinates.
    bounds : Bbox
        Bounding box of the data.
    shape : tuple of int
        Shape of the data, specified as a tuple of two integers.
    epsg : int
        EPSG code for the coordinate reference system.
    geotransform : Sequence[float]
        Sequence of geotransformation parameters.
    wavelength : float
        Radar wavelength.
    tropo_model : WeatherModel
        Model used for tropospheric correction.
    delay_type : DelayType
        Type of tropospheric delay.
    reference_file : Sequence[Filename]
        Sequence of filenames used as reference files.
    secondary_file : Sequence[Filename]
        Sequence of filenames used as secondary files.
    interferogram : str
        Identifier for the interferogram.
    reference_time : datetime.datetime
        The reference time.
    secondary_time : datetime.datetime
        The secondary time.
```

(also, let's remove the `delay_datacube` from here since the rest are all input parameters, and that field is the output)